### DISCUSSION TITLE
Add userid to stackoverflow json

### DISCUSSION THREAD

**Discussion by jayvdb** [lowlighter/metrics#630] [Ideas]
I am assuming that the markdown output format can only access variables that appear in the JSON output of the plugin.

The JSON for the stackoverflow plugin is missing the userid, so the markdown can only style the existing content.

I would like to generate a SO user link, which needs the userid.

This isnt a high priority, as I can embed the userid in the .md template, like so 

`<a href="https://stackoverflow.com/users/5037965"><img src="https://img.shields.io/badge/Reputation-{{ plugins.stackoverflow.user.reputation }}-F47F24?logo=stackoverflow&style=flat-square" alt="StackOverflow"></a>`

But then the template cant be re-used by other people.  I've used the above in SO answer https://stackoverflow.com/a/69750410/5037965 .

This might also apply to the other plugins which refer to non-Github accounts, such as `plugin_anilist_user` , `plugin_music_user` , `plugin_wakatime_user`, `plugin_tweets_user` , and others?

**Comment by lowlighter** [comment]
> I am assuming that the markdown output format can only access variables that appear in the JSON output of the plugin.

Yes indeed

___ 

Yeah these are all available data that we should expose
*(Tracked in #632)*
